### DISCUSSION TITLE
Can't install TTS with Conda and M1 Pro CPU

### DISCUSSION THREAD

**Discussion by ZachB100** [coqui-ai/TTS#1763] [General Q&A]
hey guys,
First off, I seriously love this project. I'm pretty new to neural speech synthesis, and really excited to keep going. I recently got an M1 MacBook Pro, and was somewhat disappointing to find that coqui doesn't seem to be compatible. I saw on here that someone was able to get it working using Conda, but despite installing almost every package that was available it still won't run. I'm not sure what I'm doing wrong, can someone please point me in the right direction? Luckily I still have my Intel Mac, but I would like to see this resolved as soon as possible, as those machines won't be supported forever. Thank you so much.

**Comment by domcross** [comment]
> I recently got an M1 MacBook Pro, and was somewhat disappointing to find that coqui doesn't seem to be compatible. I saw on here that someone was able to get it working using Conda, but despite installing almost every package that was available it still won't run. 

That was probably me. I cloned the TTS repo, created a new venv (with Conda), then installed all packages from requirements.txt - first with Conda, if package is not available there with pip. Some of the g2p deps (for japanese and chinese) must be installed manually (cloning repo, then buid the project). This got me so far that I can run `TTS/bin/synthesize.py` and `TTS/server/server.py`. Didn't try training, but I think even the M1 will be a bit slow for that. Now that Pytorch (nightly-builds) support the M1 GPU I will probably give it another try...

Recently I tried to install the latest TTS from PyPi (version 0.7.1, first created new venv and build the g2p deps manually - see above) but the `tts` command failed there. Didn't have the time yet to investigate...

**Reply by ZachB100** [reply]
Hey, thank you so much for your response. I attempted to build Mecab from source by doing the following. Please note that I am very new to machine learning, and not really a programmer by trade, so it's possible I'm doing this wrong. In a terminal, I moved to the repo, and ran the following command. "python setup.py -build" this began building, however I got an error about a missing Mecab config directory. Could you describe the exact steps you followed in order to build under M1? I wonder if maybe using docker would be a better idea. Again I'm pretty new to a lot of this, so apologies if this is a simple question lol.

**Reply by domcross** [reply]
I don't remember exactly how i install mecab, probably navigated to repo-folder then `pip install .`

You can also try `pip install https://github.com/SamuraiT/mecab-python3.git`, this does a temporary download of the repo, performs build/install in your environment and deletes the temp files afterwards.

**Reply by ghost** [reply]
I had to install mecab (`brew install mecab`) and then `mecab-python3` installed automatically as part of the TTS install flow.
